src/transformers/modeling_tf_utils.py

```diff
@@ -251,7 +255,7 @@ class TFSequenceClassificationLoss:
     """
 
     def hf_compute_loss(self, labels, logits):
-        if len(shape_list(logits)) == 1 or shape_list(logits)[1] == 1:
+        if logits.shape.rank == 1 or logits.shape[1] == 1:
```

Comment:
I also don't remember the exact issue, other than it often causes XLA compilation to fail :)

Reply:
I didn't see `tf.shape` used in the previous version.

Regarding `.shape` v.s `shape_list`, I am ok as long as things work. I  am just feel confused that most of the time (in other places) I see `.shape` fails while `shape_list` works (with graph mode / symbolic tensors etc.).

Reply:
Ah, sorry, I should explain! `shape_list` uses a combination of `.shape` and `tf.shape()` to build the list. I think using `tf.shape()` here confuses XLA, because it looks like a conditional that depends on the specific data you input, and those are forbidden.

I'm not 100% sure of the exact rules it uses, but all I can tell you is that it failed before and it works like this!